common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,124 @@
+"""OLI analytics service event tracker backend."""
+
+from __future__ import absolute_import
+
+import json
+import logging
+
+from requests_oauthlib import OAuth1Session
+
+from django.contrib.auth.models import User
+
+from student.models import anonymous_id_for_user
+from track.backends import BaseBackend
+
+
+LOG = logging.getLogger(__name__)
+
+
+class OLIAnalyticsBackend(BaseBackend):
+    """
+    Transmit events to the OLI analytics service
+    """
+    def __init__(self, **kwargs):
+        super(OLIAnalyticsBackend, self).__init__(**kwargs)
+
+        # the oli analytics service endpoint will be at
+        # "{}{}".format(oli_analytics_service_url, path)
+        self.oli_analytics_service_url = kwargs.get('oli_analytics_service_url', '')
+        self.path = kwargs.get('path', '')
+
+        # shared key for OLI analytics service
+        self.key = kwargs.get('key', '')
+
+        # shared secret for OLI analytics service
+        self.secret = kwargs.get('secret', '')
+
+        self.oauth = OAuth1Session(self.key, client_secret=self.secret)
+
+        # timeout for synchronous PUT attempt
+        self.timeout_in_secs = kwargs.get('timeout_in_secs', 1.0)
```

Comment:
I _love_ including the name of the unit of measurement directly in the
variable name! It makes the code much more obvious to developers in the
future.

To make this code even more obvious to future devs, I prefer to avoid
potentially ambiguous abbreviations in favor of explicitly spelling
things out; `self.timeout_in_seconds` instead of `self.timeout_in_secs`
or `self.response` instead of `self.res`